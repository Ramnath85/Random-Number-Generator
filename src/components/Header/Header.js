import React from 'react';

const Header = () => (
  <header className="App-header">
    <h1>Phone Number Generator</h1>
    <h3>Details</h3>
    <div className="sort-details">
      <p><strong>Min Phone number:</strong> 0898776765</p>
      <p><strong>Max Phone number:</strong> 0898776765</p>
      <p><strong>Total number of Phone numbers:</strong> 5</p>
    </div>
  </header>
);

export default Header;