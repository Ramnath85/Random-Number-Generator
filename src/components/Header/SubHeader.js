import React from 'react';

const SubHeader = () => (
  <h2 className="App-subtitle">
    How many numbers do you want to generate?
  </h2>
);

export default SubHeader;