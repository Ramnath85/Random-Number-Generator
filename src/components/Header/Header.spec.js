import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from "./Header";

configure({ adapter: new Adapter() });

describe('Test Header component', () => {
  let shallowComponent;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      min: null,
      max: null,
      total: 0,
      phoneNumbers: []
    };
    shallowComponent = shallow(<Header  {...defaultProps} />);
  });

  test('should render the Header component', () => {
    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.length).toEqual(1);
  });
  test('should render text', () => {
    shallowComponent.setProps({
      phoneNumbers : ['0903428765', '0787665456']
    });
    expect(shallowComponent.find('h3').text()).toContain('Details');
    expect(shallowComponent.find('strong').at(0).text()).toContain('Min Phone number');
    expect(shallowComponent.find('strong').at(1).text()).toContain('Max Phone number');
    expect(shallowComponent.find('strong').at(2).text()).toContain('Total number of Phone numbers:');
  });
});