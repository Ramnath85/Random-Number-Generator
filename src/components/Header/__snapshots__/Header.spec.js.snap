// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component should render the Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    max={null}
    min={null}
    phoneNumbers={Array []}
    total={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Phone Number Generator
        </h1>,
        false,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Phone Number Generator",
        },
        "ref": null,
        "rendered": "Phone Number Generator",
        "type": "h1",
      },
      false,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Phone Number Generator
          </h1>,
          false,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Phone Number Generator",
          },
          "ref": null,
          "rendered": "Phone Number Generator",
          "type": "h1",
        },
        false,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
